Cython/Utility/AsyncGen.c

```diff
@@ -216,15 +218,22 @@ __Pyx_async_gen_init_hooks(__pyx_PyAsyncGenObject *o)
 
     o->ag_hooks_inited = 1;
 
+#if CYTHON_COMPILING_IN_CPYTHON
     tstate = __Pyx_PyThreadState_Current;
-
     finalizer = tstate->async_gen_finalizer;
+#else
+    finalizer = _PyEval_GetAsyncGenFinalizer();
```

Comment:
These were removed from CPython a year ago: https://github.com/python/cpython/pull/32017
That probably means they are not future proof.

Reply:
The code-path is for non-CPython compilation. Is there a better way to work around the [opaque PyThreadState](https://docs.python.org/3/c-api/init.html?highlight=pythreadstate#c.PyThreadState) fields?

Reply:
> The code-path is for non-CPython compilation.

Right, but the fact that CPython had them and removed them means that other implementation won't necessarily have them either. As long as only PyPy is known to have them, I'd use the PYPY guard instead of the CPYTHON guard here.

Thefunctions were removed from CPython in favour of `sys.get_asyncgen_hooks()`, which sounds rather annoying to call from C, but we could do that here.

> Is there a better way to work around the [opaque PyThreadState](https://docs.python.org/3/c-api/init.html?highlight=pythreadstate#c.PyThreadState) fields?

It's an opaque struct in the Limited API, not in regular CPython.